File: dataset/java2python/java/FIND_TRIPLETS_ARRAY_WHOSE_SUM_EQUAL_ZERO_2.java
void f_gold ( int arr [ ] , int n ) {
  boolean found = false ;
  Arrays . sort ( arr ) ;
  for ( int i = 0 ;
  i < n - 1 ;
  i ++ ) {
    int l = i + 1 ;
    int r = n - 1 ;
    int x = arr [ i ] ;
    while ( l < r ) {
      if ( x + arr [ l ] + arr [ r ] == 0 ) {
        System . out . print ( x + " " ) ;
        System . out . print ( arr [ l ] + " " ) ;
        System . out . println ( arr [ r ] + " " ) ;
        l ++ ;
        r -- ;
        found = true ;
      }
      else if ( x + arr [ l ] + arr [ r ] < 0 ) l ++ ;
      else r -- ;
    }
  }
  if ( found == false ) System . out . println ( " No Triplet Found" ) ;
}

File: result/translate/starchat/java2python/python/FIND_TRIPLETS_ARRAY_WHOSE_SUM_EQUAL_ZERO_2.py
def f_gold(arr, n):
        arr.sort()
        found = False
        for i in range(n - 1):
            l = i + 1
            r = n - 1
            x = arr[i]
            while l < r:
                if x + arr[l] + arr[r] == 0:
                    print(x, " ", arr[l], " ", arr[r], " ")
                    l += 1
                    r -= 1
                    found = True
                elif x + arr[l] + arr[r] < 0:
                    l += 1
                else:
                    r -= 1
            if found == False:
                print("No Triplet Found")